### PR TITLE
Allow any contain call to overwrite the eagerloader.

### PR DESCRIPTION
**Pull request by markstory**
The override parameter should always work when used.

Refs #7412

### CODE REVIEW

File: src/ORM/Query.php [hunk 1]
```diff
@@ -287,7 +287,7 @@ public function eagerLoader(EagerLoader $instance = null)
      */
     public function contain($associations = null, $override = false)
     {
-        if (empty($associations) && $override) {
+        if ($override) {
             $this->_eagerLoader = null;
```
Comment (lorenzo) [inline src/ORM/Query.php]: The same eagerloader instance is used for storing `matching`. Should everything be cleared out with contain override? If not, we need another method to clear `matching` and the `joinWith`

Comment (markstory) [inline src/ORM/Query.php]: I hadn't considered that, that behavior happened before when `contain(null, true)` was used in the past. I see the concern that `contain($thing, true)` will wipe matching() calls though.

Comment (markstory) [inline src/ORM/Query.php]: Is there a way to separate containments that came from matching() / notMatching() once they've been added to the eagerloader?

Comment (lorenzo) [inline src/ORM/Query.php]: There is no way currently, but we could add a flag for this purpose.

**Comment by markstory** [comment]
@lorenzo I found a solution that I think will work. By clearing the containments and other internal state, but not resetting the _matching state we can solve the issue.

**Comment by lorenzo** [comment]
Nice!

**Pull request merged**